rs-matter/src/transport/session.rs

```diff
@@ -224,6 +223,23 @@ impl Session {
             && !self.reserved
     }
 
+    pub fn upgrade_fabric_idx(&mut self, fabric_idx: NonZeroU8) -> Result<(), Error> {
+        match &mut self.mode {
+            SessionMode::Pase { fab_idx } => {
+                if *fab_idx == 0 {
+                    *fab_idx = fabric_idx.get();
+                } else {
+                    // Upgrading a PASE session can happen only once
+                    Err(ErrorCode::Invalid)?;
+                }
+            }
+            SessionMode::Case { .. } => (), // CASE sessions are not upgradeable, as per spec
```

Comment:
Should we return an error code for this too?

Reply:
Also possible. Currently we call `upgrade_fabric_idx` unconditionally for CASE sessions too (multi-admin mode), hence the silent fall-through, but we can change the caller code to only call this method for PASE sessions. C++ SDK does it like that.

Will do.